### PR TITLE
Issue 10362: Follow up to initial service account logic

### PR DESCRIPTION
**Pull request by ayoho**
Adds logic to appropriately parse and handle the response from OpenShift's user API being used for the service account token. Also adds some `@Sensitive` annotations where we're logging the service account token

### CODE REVIEW

**Comment by ayoho** [comment]
Personal build: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_o0Qd0DYIEeqMFJAuP7DaNg

**Comment by ayoho** [comment]
Personal build passed with only a handful of unrelated failures.

File: dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/tai/AuthorizationCodeAuthenticator.java [hunk 1]
```diff
@@ -39,6 +40,7 @@
     SSLSocketFactory sslSocketFactory = null;
 
     private Map<String, Object> tokens = new HashMap<String, Object>();
+    @Sensitive
```
Comment (ezhou365) [inline dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/tai/AuthorizationCodeAuthenticator.java]: should this be in line with line 44

File: dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/internal/utils/OpenShiftUserApiUtils.java [hunk 2]
```diff
@@ -244,4 +253,63 @@ JsonObject getStatusJsonObjectFromResponse(JsonObject currentResponse) throws So
...
+
+    JsonObject addGroupsToResult(JsonObject rawJsonResponse, JsonObject metadataEntry) {
+        JsonObject result = metadataEntry;
+        String groupNameAttribute = config.getGroupNameAttribute();
+        if (groupNameAttribute != null && rawJsonResponse.containsKey(groupNameAttribute)) {
+            JsonObjectBuilder resultBuilder = copyJsonObject(metadataEntry);
+            resultBuilder.add(groupNameAttribute, rawJsonResponse.get(groupNameAttribute));
+            result = resultBuilder.build();
+        }
+        return result;
```
Comment (ezhou365) [inline dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/internal/utils/OpenShiftUserApiUtils.java]: does it matter if group is added to the end or beginning of the jsonObject object

**Pull request merged**